libs/webb-ui-components/src/components/ListCard/ContractListCard.tsx

```diff
@@ -0,0 +1,114 @@
+import { useState, useMemo, forwardRef } from 'react';
+import { ExternalLinkLine, Search } from '@webb-tools/icons';
+import { ScrollArea } from '../ScrollArea';
+import { Typography } from '../../typography';
+import SkeletonLoader from '../SkeletonLoader';
+import { Input } from '../Input';
+import { ListItem } from './ListItem';
+import { ListCardWrapper } from './ListCardWrapper';
+import { isHex } from 'viem';
+
+import { ContractListCardProps } from './types';
+import { shortenHex, shortenString } from '../../utils';
+
+const ContractListCard = forwardRef<HTMLDivElement, ContractListCardProps>(
+  ({ isLoading, selectContractItems, ...props }, ref) => {
+    const [searchText, setSearchText] = useState('');
+
+    const filterList = useMemo(
+      () =>
+        selectContractItems.filter((item) => {
+          return (
+            item.name.toLowerCase().includes(searchText.toLowerCase()) ||
+            item.address.toLowerCase().includes(searchText.toLowerCase()) ||
+            item.blockExplorerUrl
+              ?.toLowerCase()
+              .includes(searchText.toLowerCase())
+          );
+        }),
+      [selectContractItems, searchText]
+    );
+
+    return (
+      <ListCardWrapper
+        title="Select Contract"
+        hideCloseButton
+        {...props}
+        ref={ref}
+      >
+        {/** The search input */}
+        <div className="py-4">
+          <Input
+            id="chain"
+            rightIcon={<Search />}
+            placeholder="Search chains"
```

Comment:
I think it's not correct to "Search chains", rather it should be searching contracts. Remember that all the contracts that can be governed exist on all the chains.